src/components/Gallery/ImagePagination.tsx

```diff
@@ -32,12 +33,14 @@ export function ImagePagination({
       changePage(nextPage);
     }
   };
+  const debouncePaginateForward = debounce(paginateForward, 1000);
   const paginateBackward = () => {
     const prevPage = pageInfo.currentIndex - 1;
     if (prevPage >= 0) {
       changePage(prevPage);
     }
   };
+  const debouncePaginateBackward = debounce(paginateBackward, 1000);
```

Comment:
I had originally added `{ leading: true }` to these calls because I thought it made for a slightly smoother experience. But for some reason adding it does not work and in fact, it will cause debouncing stops working altogether. I'm scratching my head as to why that could be so feel free to add in any ideas I might have missed

Reply:
hmm, if I had to guess it could have something to do with the `paginateBackwards` fn itself. Maybe this doesn't work as expected if your fn call a fn passed down from props?

But yes, that's bizarre.

Reply:
I _think_ you were onto something here. I tried debouncing the fn that's responsible for changing the page state instead of the `onClick` handler. It seems to work better now IMO.